x-pack/libbeat/management/managerV2.go

```diff
@@ -893,6 +893,13 @@ func (cm *BeatV2Manager) reloadAPM(unit *agentUnit) {
 			apmConfig = expected.APMConfig
 		}
 	}
+
+	if (cm.lastAPMCfg == nil && apmConfig == nil) || (cm.lastAPMCfg != nil && gproto.Equal(cm.lastAPMCfg, apmConfig)) {
```

Comment:
[To reviewer] With this change, `apm.Reload(nil)` is never called if apmConfig is always nil, even when starting the beat. I assume that is fine.

Reply:
Skipping the `apm.Reload(nil)` would potentially skip setting the initial configuration to an empty one to disable tracing. The implementation of this is on the APM server side. https://github.com/elastic/apm-server/blob/840fe6c8cb328bbb07ce49c21d6a3068be64d9e4/internal/beatcmd/reloader.go#L213-L227

This depends on what the default state is, or if it would ever change. If the default state is that APM tracing is disabled the initial `apm.Reload(nil)` has no effect. If it isn't disabled then skipping it has a consequence.

Reply:
Confirmed that apm-server is the only beat that uses apm tracing config reload. As apm-server doesn't rely on the initial nil reload and defaults to no trace, it is fine to skip the initial nil reload.